torch/_dynamo/external_utils.py

```diff
@@ -249,3 +249,25 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> _R:
                 return fn(*args, **kwargs)
 
     return wrapper
+
+
+def filter_out_const_values(tup: tuple[Any, ...], mask: list[bool]) -> tuple[Any, ...]:
+    out = []
+    for mask_idx, m in enumerate(mask):
+        if not m:
+            out.append(tup[mask_idx])
+    return tuple(out)
+
+
+def insert_const_values_with_mask(
+    tup: tuple[Any, ...], mask: list[bool], values: tuple[Any, ...]
```

Comment:
so, mask is always the same length as values? Is there a filler value inside `values` when the mask is False at some elements?

If there are invariants then let's assert them somewhere